client/lib/domains/reducer.js

```diff
@@ -82,6 +83,16 @@ function reducer( state, payload ) {
 			} );
 
 		case UpgradesActionTypes.PRIMARY_DOMAIN_SET_COMPLETED:
+			return updateSiteState( state, siteId, {
```

Comment:
I guess you're right - kinda like we shouldn't add unnecessary things to a component's state.
OK, so, instead, what do you think about changing:
```
if ( action.domainName === domain.name ) {
	return Object.assign( {}, domain, { isPrimary: true } );
}
return Object.assign( {}, domain, { isPrimary: false } );
```
to just:
```
return Object.assign( {}, domain, { isPrimary: action.domainName === domain.name } );
```